File: backend/src/main/java/me/sarismart/backend/Entity/Store.java
package me.sarismart.backend.Entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.util.List;
import java.util.ArrayList;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "stores")
public class Store {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, unique = true)
    private String storeName;
    
    @Column(nullable = false)
    private String location;

    @Column(nullable = false)
    private double latitude;

    @Column(nullable = false)
    private double longitude;

    @ManyToOne
    @JoinColumn(name = "owner_id", nullable = false)
    private User storeOwnerId;

    @OneToMany
    private List<User> workers = new ArrayList<>();

    @JsonIgnore
    @OneToMany(mappedBy = "store", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Product> products = new ArrayList<>();

    @OneToMany(mappedBy = "store", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Sale> sales = new ArrayList<>();
}

File: frontend_mobile/app/src/main/java/edu/cit/sarismart/features/user/tabs/stores/data/models/Store.kt
package edu.cit.sarismart.features.user.tabs.stores.data.models

import edu.cit.sarismart.features.user.tabs.account.data.models.User

data class Store (
    val id: Long,
    val storeName: String,
    val location: String,
    val latitude: Double,
    val longitude: Double,
    val owner: User,
    val workers: List<User>,
    val sales: List<Sale>?
)